src/content/en/fundamentals/media/fast-playback-with-video-preload.md

```diff
@@ -0,0 +1,559 @@
+
+For the sake of simplicity, let's assume the entire video has been split into
+smaller files like "file_1.webm", "file_2.webm", "file_3.webm", etc. We'll use
+MSE to feed the video element with those once the `onload` event from `<link
+rel="preload">` has been fired.
+
+```
+<link rel="preload" as="video" href="https://cdn.com/file_1.webm"
+    onload="preloadFinished()" onerror="preloadFailed()">
+
+<video id="video" controls></video>
+
+<script>
+  // This function is automatically executed when the first segment of video is
+  // preloaded successfully. We may call it explicitely as well later on.
+  function preloadFinished() {
+    const mediaSource = new MediaSource();
+    video.src = URL.createObjectURL(mediaSource);
+    mediaSource.addEventListener('sourceopen', sourceOpen);
+  }
+
+  function sourceOpen(event) {
+    URL.revokeObjectURL(video.src);
+    const mediaSource = event.target;
+    const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
+
+    // If video is preloaded already, fetch will return immediately a response
+    // from the browser cache (memory cache). Otherwise, it will perform a
+    // regular network fetch.
+    fetch('https://cdn.com/file_1.webm')
+    .then(response => response.arrayBuffer())
+    .then(data => {
+      // Append the data into the new sourceBuffer.
+      sourceBuffer.appendBuffer(data);
+      // TODO: Fetch file_2.webm when user starts playing video.
+    })
+    .catch(error => {
+      // TODO: Show "Video is not available" message to user.
+    });
+  }
+
+  // This function is called if the first segment of video is not available.
+  function preloadFailed() {
+    // TODO: Show "Video is not available" message to user.
+  }
+</script>
+```
+
+Warning: For cross-origin resources, make sure your CORS headers are set
+properly as opaque responses retrieved with `fetch(videoFileUrl, { mode:
+'no-cors' })` are not allowed to feed any video or audio element.
+
+### Support
+
+As it is not supported in every browser yet, you may want to detect its
+availability with the snippet below.
+
+```
+<link rel="preload" as="video" href="https://cdn.com/file_1.webm"
+    onload="preloadFinished()" onerror="preloadFailed()">
+
+<video id="video" controls></video>
+
+<script>
+  if (!preloadVideoSupported()) {
+    // TODO: Fetch video from the network and load it.
+  }
+
+  function preloadFinished() {
+    // TODO: Load video.
+  }
+
+  function preloadFailed() {
+    // TODO: Show "Video is not available" message to user.
+  }
+
+  function preloadVideoSupported() {
+    const link = document.createElement('link');
+    link.as = 'video';
+    return (link.as === 'video');
+  }
+</script>
+```
+
+## Manual Buffering
+
+Before we dive into the [Cache API] and service workers, let's see how to
+manually buffer a video with MSE. The example below assumes that your web
+server supports HTTP Range requests. Note that this would be pretty similar to
+segment files.
```

Comment:
Maybe note that there are entire middleware libraries dedicated to this; shaka, jwplayer, video.js, etc.